src/gatsby/models/authors/parse-link.js

```diff
@@ -1,31 +1,77 @@
-const processStringLink = require('./string-link-processor.js')
+/*
+   Processor builder: makes a function that turns a string into processed Link metadata
+
+   Takes "Mod functions" as parameters, returns a function that starts with {} and pipes
+   that through each function, then returns the result.
+
+   signature: ({ scheme: string, host: string, pathname: string}, input)
+   example:      ^ https://      ^ mysite.com  ^ /rest/of/the/url ^ an object
+
+   The second parameter will be the current state of the input, and usually will
+   have to be spread into your return result.
+ */
+function processor(...mods) {
+  return function process(groups) {
+    return mods.reduce(
+      (currentResult, _, i) => mods[i](groups, currentResult),
+      {}
+    )
+  }
+}
+
+// Processor helpers
+
+function trimSlashes(input) {
+  return /^\/?(.*?)\/?$/.exec(input)[1]
+}
+
+const asSite = site => (groups, input) => ({
+  ...input,
+  site
+})
+
+const pathnameAsUsername = ({ pathname }, input) => ({
+  ...input,
+  username: trimSlashes(pathname)
+})
+
+const urlHTTPS = ({ host, pathname }, input) => ({
+  ...input,
+  url: 'https://' + host + pathname
```

Comment:
where do we get `/` between host and path?

Reply:
That's just how the regex breaks up the string now. I used to strip the leading slash in the regex, but it involved wrapping the pathname in a non-capturing group just to strip a slash and then have it be re-added here later.

I decided to simplify the regex and just let the processors strip the leading slash when they need to. Not only is the regex cleaner, but more operations use the pathname with the slash than without currently- most obviously these url functions that are run on every parsed string.

Reply:
I see. I think this is good. I think it's consistent even with URI (or how is it named in JS?) class.